pkg/kubelet/kubelet_pods.go

```diff
@@ -86,24 +86,115 @@ func (kl *Kubelet) GetActivePods() []*v1.Pod {
 	return activePods
 }
 
+func (kl *Kubelet) parseDeviceSpec(deviceSpec string) (deviceInfo kubecontainer.DeviceInfo, err error) {
+	if len(deviceSpec) == 0 {
+		err = fmt.Errorf("device spec is empty")
+		return
+	}
+
+	deviceArr := strings.SplitN(deviceSpec, ":", 3)
+	if !filepath.IsAbs(deviceArr[0]) {
+		err = fmt.Errorf("incorrect value of PathOnHost in device spec: %s", deviceSpec)
+		return
+	}
+
+	// PathOnHost
+	deviceInfo.PathOnHost = deviceArr[0]
+
+	// PathInContainer
+	if len(deviceArr) >= 2 && filepath.IsAbs(deviceArr[1]) {
+		deviceInfo.PathInContainer = deviceArr[1]
+	} else {
+		deviceInfo.PathInContainer = deviceInfo.PathOnHost
+	}
+
+	// Permissions
+	if len(deviceArr) == 3 && len(deviceArr[2]) <= 3 && strings.ContainsAny(deviceArr[2], "mrw") {
```

Comment:
Should there be some some more validation on what the user passes here? I don't know if all of the defaults are sufficient and it might be better to log and reject.